File: 8.longest_common_sequence/main.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    let count = 0;
    let lastSmall = Number.MIN_SAFE_INTEGER;
    let longestSeq = 1;

    nums.sort((a, b) => a - b);
    for(let i=0; i<nums.length; i++){
        if(lastSmall+1 !== nums[i] && lastSmall !== nums[i]){
            count = 1;
            lastSmall = nums[i];
        }
        if(lastSmall+1 === nums[i]){
            count++;
            lastSmall = nums[i];
        }
        longestSeq = Math.max(longestSeq, count);
    }
    return longestSeq;
};

File: 8.longest_common_sequence/main.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length == 0) return 0;

        int count = 0;
        int lastSmall = Integer.MIN_VALUE;
        int longestSeq = 1;

        Arrays.sort(nums);
        for(int i=0; i<nums.length; i++){
            if(lastSmall+1 != nums[i] && lastSmall != nums[i]){
                count=1;
                lastSmall = nums[i];
            }
            if(lastSmall+1 == nums[i]){
                count++;
                lastSmall = nums[i];
            }
            longestSeq = Math.max(longestSeq, count);
        }
        return longestSeq;
    }
}